From: Marcin Serwin <marcin@serwin.dev>
Date: Fri, 9 Jan 2026 19:42:59 +0100
Subject: test: fix incorrect pointer casts

When building with cmocka 2.0.1 the tests fail due to stricter type
checks, this patch fixes that. Tested with both cmocka 1.1.8 and 2.0.1.

```
test/unit/tcti-libtpms.c:148:5: error: initialization of 'const void *' from 'enum TPMLIB_StateType' makes pointer from integer without a cast [-Wint-conversion]
  148 |     check_expected_ptr(st);
      |     ^~~~~~~~~~~~~~~~~~
test/unit/tcti-libtpms.c:150:5: error: initialization of 'const void *' from 'uint32_t' {aka 'unsigned int'} makes pointer from integer without a cast [-Wint-conversion]
  150 |     check_expected_ptr(buf_len);
      |     ^~~~~~~~~~~~~~~~~~
```


Signed-off-by: Marcin Serwin <marcin@serwin.dev>
---
 test/unit/tcti-libtpms.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/test/unit/tcti-libtpms.c b/test/unit/tcti-libtpms.c
index 11e501a..fc3c106 100644
--- a/test/unit/tcti-libtpms.c
+++ b/test/unit/tcti-libtpms.c
@@ -145,9 +145,9 @@ TPM_RESULT TPMLIB_Process(unsigned char **resp_buf, uint32_t *resp_len, uint32_t
 }
 TPM_RESULT TPMLIB_SetState(enum TPMLIB_StateType st, const unsigned char *buf, uint32_t buf_len)
 {
-    check_expected_ptr(st);
+    check_expected(st);
     check_expected_ptr(buf);
-    check_expected_ptr(buf_len);
+    check_expected(buf_len);
     return mock_type(int);
 }
 void TPMLIB_Terminate(void)
